### PR TITLE
feat: rock only 519 

### PR DESCRIPTION
**Pull request by clemens-tolboom**
## Provides

- [x] rock only images for
  - [x] `release-test.R`
  - [x] `application.yml` default profile

## Needs

- [x] remove java code regarding 'rserver'
- [x] remove java code selecting correct service 'rock' or 'rserve'
- [x] 3 tests are disabled .. needs review how to fix (We switched to Rock which were not tests previously so I tried to fix those but failed.)
- [x] fix docker checks not trying to connect when server is not ready?

### CODE REVIEW

**Comment by clemens-tolboom** [comment]
## Tests

### armadillo/test

./gradlew :armadillo:test

```

ArmadilloUtilsTest > testCreateRawResponse() FAILED
    org.opentest4j.AssertionFailedError: array lengths differ, expected: <3> but was: <38>
        at app//org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
        at app//org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
        at app//org.junit.jupiter.api.AssertArrayEquals.assertArraysHaveSameLength(AssertArrayEquals.java:428)
        at app//org.junit.jupiter.api.AssertArrayEquals.assertArrayEquals(AssertArrayEquals.java:205)
        at app//org.junit.jupiter.api.AssertArrayEquals.assertArrayEquals(AssertArrayEquals.java:63)
        at app//org.junit.jupiter.api.AssertArrayEquals.assertArrayEquals(AssertArrayEquals.java:59)
        at app//org.junit.jupiter.api.Assertions.assertArrayEquals(Assertions.java:1226)
        at app//org.molgenis.armadillo.controller.ArmadilloUtilsTest.testCreateRawResponse(ArmadilloUtilsTest.java:42)
```

RExpRaw on byte array becomes String of class name array length `String "org.rosuda.REngine.REXPRaw@3a3f96ab[3]"`

### r/test

```
RExecutorServiceImplTest > testInstallPackage() FAILED
    org.molgenis.r.exceptions.FailedRPackageInstallException: The package [ 'location/_test' ] is not installed correctly. It could be a dependency problem.
        at app//org.molgenis.r.service.RExecutorServiceImpl.installPackage(RExecutorServiceImpl.java:157)
        at app//org.molgenis.r.service.RExecutorServiceImplTest.testInstallPackage(RExecutorServiceImplTest.java:266)
```
Jumps into `!result.asLogical()` somehow

https://github.com/molgenis/molgenis-service-armadillo/blob/3b12dec222280fc0896f5cc2a713cf7539257d94/r/src/main/java/org/molgenis/r/service/RExecutorServiceImpl.java#L155-L158

**Comment by clemens-tolboom** [comment]
## Disabled R tests

1. https://github.com/molgenis/molgenis-service-armadillo/blob/ef6c65adb6fef054974c396f0bf6f8d0e3e663df/r/src/test/java/org/molgenis/r/service/RExecutorServiceImplTest.java#L253-L276
2. @Test
  @Disabled
  void testCountRserveProcesses() throws REXPMismatchException, RServerException {
    when(rExecutorService.execute(COUNT_RSERVE_PROCESSES_COMMAND, rConnection))
        .thenReturn(new RockResult(new REXPList(new REXPInteger(3), "n")));
    assertEquals(3, processService.countRserveProcesses(rConnection));
  }

because RockResult does not know how to process as response RExpr like `REXPList`, `REXPNull` or `REXPLogical` but JSON.

see ie https://rockdoc.obiba.org/en/latest/rest/session.html#evaluate

Guess this needs a rewrite.

**Comment by marikaris** [comment]
[M10337 As Armadillo user I want to be able to use rock in production](https://trello.com/c/YBe8KSUV/1092-m10337-as-armadillo-user-i-want-to-be-able-to-use-rock-in-production)

**Comment by clemens-tolboom** [comment]
### Tests

We switched to Rock which were not tests previously so I tried to fix those but failed.

File: r/src/main/java/org/molgenis/r/RServerConnectionFactory.java [hunk 1]
```diff
@@ -15,17 +16,45 @@ public RServerConnectionFactory(EnvironmentConfigProps environment) {
     this.environment = environment;
   }
 
-  @Override
-  public RServerConnection tryCreateConnection() {
+  int doHead(String uri) {
```
Comment (mswertz) [inline r/src/main/java/org/molgenis/r/RServerConnectionFactory.java:19]: why change 'public' to be friendly ?

File: r/src/test/java/org/molgenis/r/service/ProcessServiceImplTest.java [hunk 2]
```diff
@@ -56,9 +57,19 @@ void testTerminate() throws RServerException {
   }
 
   @Test
+  @Disabled
```
Comment (mswertz) [inline r/src/test/java/org/molgenis/r/service/ProcessServiceImplTest.java:60]: why disable test?

File: armadillo/src/main/resources/application.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ armadillo:
 
   profiles:
     - name: default
-      image: datashield/armadillo-rserver
+      image: datashield/rock-base:latest
```
[the remark this answers is not in the document]
Comment (clemens-tolboom) [reply armadillo/src/main/resources/application.yml:11]: When `/data/system/profiles.json` is deleted this profile is used.

File: r/src/test/java/org/molgenis/r/service/ProcessServiceImplTest.java (hunk 2 above)
Comment (clemens-tolboom) [reply r/src/test/java/org/molgenis/r/service/ProcessServiceImplTest.java:60]: I did not manage to convert its Rock stub ... see FIXME comment below

File: r/src/main/java/org/molgenis/r/RServerConnectionFactory.java (hunk 1 above)
Comment (clemens-tolboom) [reply r/src/main/java/org/molgenis/r/RServerConnectionFactory.java:19]: Not sure I understand ... should `int doHead()` be public?

**Review by marikaris — CHANGES_REQUESTED** [review]

File: r/src/test/java/org/molgenis/r/service/ProcessServiceImplTest.java (hunk 2 above)
Comment (clemens-tolboom) [reply r/src/test/java/org/molgenis/r/service/ProcessServiceImplTest.java:60]: Discussed: we'll fix those after merging and think need to

File: r/src/main/java/org/molgenis/r/RServerConnectionFactory.java (hunk 1 above)
Comment (clemens-tolboom) [reply r/src/main/java/org/molgenis/r/RServerConnectionFactory.java:19]: I see now ... it was already public ... guess this is resolved?

[the remark this answers is not in the document]
Comment (clemens-tolboom) [reply r/src/main/java/org/molgenis/r/RServerConnectionFactory.java:19]: Reviewed together

File: scripts/release/release-test.R [hunk 4]
```diff
@@ -49,9 +49,9 @@ ADMIN_MODE <- FALSE
 # default profile settings in case a profile is missing
 profile_defaults = data.frame(
   name = c("xenon", "rock"),
-  container = c("datashield/armadillo-rserver_caravan-xenon:1.0.0", "datashield/rock-base:latest"),
-  port = c("", 8085),
-  # Multiple packages can be concattenated using ,, then using stri_split_fixed() to break them up again
+  container = c("datashield/rock-base:latest", "datashield/rock-base:latest"),
```
[the remark this answers is not in the document]
Comment (clemens-tolboom) [reply scripts/release/release-test.R:52]: TNX ... done

**Comment by clemens-tolboom** [comment]
It seems `datashield/rock-base` works and `datashield/rock-dolomite-xenon:latest` does not ... needs investigation

**Comment by StuartWheater** [comment]
I will check, but I think datashield/rock-dolomite-xenon was based on the datashield/rock-base of the time, and new a version of rock-base has been generated.

**Review by marikaris — APPROVED** [review]

**Pull request merged**